src/actions/receive-profile.js

```diff
@@ -910,8 +914,13 @@ export function retrieveProfileFromFile(
           // extensions (eg .profile). So we can't rely on the mime type to
           // decide how to handle them. We'll try to parse them as a plain JSON
           // file.
-          const text = await fileReader(file).asText();
-          const profile = await unserializeProfileOfArbitraryFormat(text);
+          let profile;
+          if (isInstrumentsProfile(file)) {
```

Comment:
also, it would be cool to check for it as a zip file as well (but then we can't check the filename only, we need to check the content). That can wait for a future PR though.

Reply:
Also, I can't make it work with the button "load as file" on the homepage, maybe we need to accept directories in the file upload control.

Reply:
Regarding `file.type`, we are not getting `undefined` as `file.type` for Instruments profiles. That's why I put it into a default case.

For zip file, yes we can add support for it :)

In last comment, when we hit a `load as file` button and try to upload a Instruments file, it looks like this.
<img width="800" alt="Screenshot 2019-08-31 at 2 59 38 AM" src="https://user-images.githubusercontent.com/16744065/64052311-a891af00-cb9b-11e9-810f-5c7dbd3a5916.png">
 
So, yes we will need to add support for directory upload.